src/middlewared/middlewared/plugins/ups.py

```diff
@@ -31,6 +31,8 @@ class Config:
         data['mode'] = data['mode'].upper()
         data['shutdown'] = data['shutdown'].upper()
         data['toemail'] = [v for v in data['toemail'].split(';') if v]
+        host = 'localhost' if data['mode'] == 'MASTER' else data['remotehost']
+        data['complete_identifier'] = f'{data["identifier"]}@{host}:{data["remoteport"]}'
```

Comment:
IIRC correctly if you don't pop this in do_update, it will raise an exception?

Reply:
It certainly will and i have made the change locally : ) Testing master/slave configuration right now and will push this change and any other if required : )